broker-core/src/main/java/io/zeebe/broker/clustering/base/partitions/Partition.java

```diff
@@ -72,14 +74,18 @@ public Partition(
 
   @Override
   public void start(final ServiceStartContext startContext) {
+    final boolean isSingleBroker = brokerCfg.getCluster().getClusterSize() == 1;
```

Comment:
I guess we should check the replication factor instead. It could be that we have a cluster count > 1 but replication factor == 1.